plugins/modules/win_acl_inheritance.ps1

```diff
@@ -1,75 +1,156 @@
-                        ($inheritedRule.IdentityReference -eq $explicitRrule.IdentityReference) -And
-                        ($inheritedRule.InheritanceFlags -eq $explicitRrule.InheritanceFlags) -And
-                        ($inheritedRule.PropagationFlags -eq $explicitRrule.PropagationFlags)
-                    ) {
-                        $objACL.RemoveAccessRule($explicitRrule)
+    Process {
+        $properties = $acl.Access |
+            Get-Member -MemberType Property |
+            Where-Object { $_.Name -ne 'IsInherited' } |
+            Select-Object -ExpandProperty Name
+
+        ForEach ($inheritedRule in $($acl.Access | Where-Object { $_.IsInherited })) {
+            ForEach ($explicitRule in $($acl.Access | Where-Object { -not $_.IsInherited })) {
+                If ($null -eq (Compare-Object -ReferenceObject $explicitRule -DifferenceObject $inheritedRule -Property $properties)) {
+                    if ($PSCmdlet.ShouldProcess($acl)) {
+                        $acl.RemoveAccessRule($explicitRule)
+                    }
+                    else {
+                        Write-Verbose "WhatIf: Performing remove explicit duplicate rule: $explicitRule"
                     }
                 }
             }
         }
+        return $acl
+    }
+
+    End {}
+}
 
-        Set-ACL -LiteralPath $path -AclObject $objACL -WhatIf:$check_mode
-        $result.changed = $true
+$module = [Ansible.Basic.AnsibleModule]::Create($args, $spec)
+
+$path = $module.Params.path
+$reorganize = $module.Params.reorganize
+$state = $module.Params.state
+$check_mode = $module.CheckMode
+
+$module.Result.changed = $false
+
+$pathQualifier = Split-Path -Path $path -Qualifier -ErrorAction SilentlyContinue
+
+if ($pathQualifier -eq 'HKCR:' -and (-not (Test-Path -LiteralPath HKCR:\))) {
+    $null = New-PSDrive -Name HKCR -PSProvider Registry -Root HKEY_CLASSES_ROOT
+}
+if ($pathQualifier -eq 'HKU:' -and (-not (Test-Path -LiteralPath HKU:\))) {
+    $null = New-PSDrive -Name HKU -PSProvider Registry -Root HKEY_USERS
+}
+if ($pathQualifier -eq 'HKCC:' -and (-not (Test-Path -LiteralPath HKCC:\))) {
+    $null = New-PSDrive -Name HKCC -PSProvider Registry -Root HKEY_CURRENT_CONFIG
+}
+
+If (-Not (Test-Path -LiteralPath $path)) {
+    $module.FailJson("$path does not exist")
+}
+
+try {
+    $acl = Get-AclEx -LiteralPath $path
+}
+catch {
+    $module.FailJson($_.ToString())
+}
+
+$module.Diff.before = @{ access_rules_protected = $acl.AreAccessRulesProtected }
+
+If (($state -eq 'present') -and $acl.AreAccessRulesProtected) {
+    try {
+        $acl.SetAccessRuleProtection($false, $false)
     }
-    Elseif (($state -eq "absent") -And (-not $inheritanceDisabled)) {
-        $objACL.SetAccessRuleProtection($True, $reorganize)
-        Set-ACL -LiteralPath $path -AclObject $objACL -WhatIf:$check_mode
-        $result.changed = $true
+    catch {
+        $module.FailJson($_.ToString())
```

Comment:
Same thing here, include a brief error message that describes the action and also pass in $_ as the 2nd argument.

You can also just omit the whole try/catch block. An uncaught exception will be displayed with the exception details on higher verbosities as well.